File: app/src/main/java/de/xikolo/storages/base/BaseStorage.java
package de.xikolo.storages.base;

import android.content.SharedPreferences;

import de.xikolo.App;

public abstract class BaseStorage {

    protected SharedPreferences preferences;

    public BaseStorage(String name, int mode) {
        this.preferences = App.getInstance().getSharedPreferences(name, mode);
    }

}

File: app/src/main/java/de/xikolo/storages/base/BaseStorage.kt
package de.xikolo.storages.base

import android.content.SharedPreferences

import de.xikolo.App

abstract class BaseStorage(name: String, mode: Int) {

    private val preferences: SharedPreferences = App.getInstance().getSharedPreferences(name, mode)

    fun getString(key: String, defValue: String? = null) : String? = preferences.getString(key, defValue)

    fun putString(key: String, value: String?) {
        preferences
            .edit()
            .putString(key, value)
            .commit()
    }

    fun delete() {
        preferences
            .edit()
            .clear()
            .apply()
    }

}
